### PR TITLE
Feature/sr 11 implement featured products section

### PR DESCRIPTION
**Pull request by GalacticOctopus**
+ featured page
+ functional slider (mobile responsive)

### CODE REVIEW

**Comment by GalacticOctopus** [comment]
Three commits are from previous branch because I had to merge them. Please ignore them

File: src/components/Slider.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    if (!carouselRef.current) return;
+    event.preventDefault();
+
+    // when we stop holding mouse button stop dragging
+    if (!isDragStart) {
+      return;
+    }
+
+    setIsDragging(true);
+    carouselRef.current.classList.add("dragging");
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:63]: Change class name conditionally in JSX based on state instead modifying DOM like that.

File: src/components/Slider.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    setIsDragStart(false);
+    if (carouselRef.current) carouselRef.current.classList.remove("dragging");
+    if (!isDragging) return;
+    setIsDragging(false);
+
+    // cleanup
+    setPositionDiff(0);
+  };
+
+  function handleTouchDown(event: React.TouchEvent<HTMLDivElement>) {
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:80]: Import this interface directly, not from `React`.

File: src/components/Slider.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        {value.map((v) => (
+          <div className="featured__product">
+            <img src={v.imageSource} alt="Electric scooter" draggable="false" />
+
+            <div className="featured__product--description">
+              <h2 className="text-700 text-body-sm">{v.title}</h2>
+              <p className="text-caption">{v.caption}</p>
+              <a href="/reservation" className="text-caption link-underlined">
+                Make a reservation
+                <img src="src\assets\arrow-right.svg" alt="Arrow right" />
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:135]: Import this image

File: src/components/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          <img src={arrowLeft} alt="Arrow left" />
+        </span>
+        {value.map((v) => (
+          <div className="featured__product">
+            <img src={v.imageSource} alt="Electric scooter" draggable="false" />
+
+            <div className="featured__product--description">
+              <h2 className="text-700 text-body-sm">{v.title}</h2>
+              <p className="text-caption">{v.caption}</p>
+              <a href="/reservation" className="text-caption link-underlined">
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:133]: Use `react-router` links

File: src/components/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        <span
+          id="left"
+          ref={arrowLeftRef}
+          onClick={() => {
+            if (carouselRef.current) carouselRef.current.scrollLeft -= 325;
+          }}
+        >
+          <img src={arrowLeft} alt="Arrow left" />
+        </span>
+        {value.map((v) => (
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:126]: Add unique id to each product and use it as value for `key` prop. Also use meaningful names, not `v`, `e` etc.

File: src/components/Slider.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+              </a>
+            </div>
+          </div>
+        ))}
+
+        <span
+          id="right"
+          ref={arrowRightRef}
+          onClick={() => {
+            if (carouselRef.current) carouselRef.current.scrollLeft += 325;
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:145]: Create function for this operation.

File: src/pages/Featured.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      "The PRO-II is the most technically advanced scooter in the Ducati Urban e-mobility series.",
+    imageSource: "src/assets/product1.png",
+  },
+  {
+    title: "Ducati Pro - IV Electric Scooter",
+    caption:
+      "The PRO-II is the most technically advanced scooter in the Ducati Urban e-mobility series.",
+    imageSource: "src/assets/product1.png",
+  },
+];
```
Comment (masterHAWK99) [inline src/pages/Featured.tsx:29]: Add real products

File: src/pages/Featured.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <div className="featured">
+      <div className="featured__left">
+        <div className="featured__wrapper">
+          <h2 className="text-heading2">Featured Vehicles</h2>
+          <p className="text-body">
+            These are recently the most <br /> frequently chosen and most valued{" "}
+            <br />
+            vehicles by our customers around the <br /> world.
+          </p>
```
Comment (masterHAWK99) [inline src/pages/Featured.tsx:41]: Get rid of `br`'s

File: src/components/Slider.tsx [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+          }}
+        >
+          <img src={arrowLeft} alt="Arrow left" />
+        </span>
+        {value.map((v) => (
+          <div className="featured__product">
+            <img src={v.imageSource} alt="Electric scooter" draggable="false" />
+
+            <div className="featured__product--description">
+              <h2 className="text-700 text-body-sm">{v.title}</h2>
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:131]: Use correct styling for title

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

File: src/components/Slider.tsx [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+  caption: string;
+  imageSource: string;
+  id: number;
+};
+
+type SliderProps = {
+  value: SliderElement[];
+};
+
+export function Slider({ value }: SliderProps) {
```
Comment (masterHAWK99) [inline src/components/Slider.tsx:25]: Please move logic code to hook

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

**Review by masterHAWK99 — APPROVED** [review]

**Pull request merged**